Exercises/NoModules/Chapter 16/Soln16_04/Soln16_04.cpp

```diff
@@ -0,0 +1,92 @@
+// Throwing and catching standard exceptions
+#include <iostream>
+#include <stdexcept>
+#include <vector>
+#include <format>
+#include <typeinfo>
+#include <optional>
+
+/*
+  This solution triggers all exceptions mentioned in the text 
+  accompanying the Table with all Standard Library exceptions, in order.
+  To know how to trigger any other type, 
+  you will have to consult your Standard Library reference.
+
+  Note: while you'll typically catch exceptions by reference-to-const-std::exception 
+  (unless more specific handling for concrete types is required, of course),
+  we specify the concrete exception type instead here for clarity.
+ */
+
+// First, some dummy class types...
+class BaseClass
+{
+public:
+  virtual ~BaseClass() = default;
+};
+
+class DerivedClass1 : public BaseClass {};
+class DerivedClass2 : public BaseClass {};
+
+int main()
+{
+  try
+  {
+    std::vector v{ 1, 2, 3, 4, 5 };
+    std::cout << v[10] << std::endl;
```

Comment:
this will never throw out_of_range.
You need to use at().

Reply:
Oopsie...